selfservice/flow/registration/handler.go

```diff
@@ -354,6 +366,16 @@ func (h *Handler) fetchFlow(w http.ResponseWriter, r *http.Request, ps httproute
 		return
 	}
 
+	if ar.HydraLoginChallenge.Valid {
```

Comment:
Please add tests for this in handler_test.go

If you don't want to boot hydra, you could also use a `gomock` implementation of the hydra client